Python/gc.c

```diff
@@ -166,12 +166,31 @@ _PyGC_InitState(GCState *gcstate)
 #undef INIT_HEAD
 }
 
+static void
+gc_set_threshold_from_env(PyInterpreterState *interp)
+{
+    const PyConfig *config = _PyInterpreterState_GetConfig(interp);
+    const char *env = _Py_GetEnv(config->use_environment,
+                                 "PYTHON_GC_THRESHOLD");
+    if (env == NULL || strcmp(env, "default") == 0) {
+        return;
+    }
+    int threshold = -1;
+    if (_Py_str_to_int(env, &threshold) < 0) {
+        return; // parse failed, silently ignore
+    }
```

Comment:
Hmm, why ignore the error? If a user accidentally sets the variable to something that doesn't work, then they might end up very confused as to why Python is ignoring their request.

Reply:
I'm hoping this change can be included in the 3.13 RC and so I wanted it as simple and robust as possible.  In the 'main' branch, I plan to re-work it so it becomes part of the `config` structure and I would add an error raise there.

Maybe it is safe enough to raise an error here too.  That would be consistent with the principle that "errors should not pass silently".

Reply:
I think that making it return an error rather than nothing should still remain fairly simple, `return` will just become `return _PyStatus_ERR("...")` and whatnot.

I haven't paid too much attention to the incremental GC problem (#124567), but I'm assuming that this is a possible solution for it? That should help it go into the 3.13 release

Reply:
No, I don't think this would be enough for us to feel safe about including incremental GC into 3.13.  In retrospect, the incremental GC should have been opt-in for 3.13 and would become default in 3.14 if people had good experience with it.

The point of this env var is to increase the chances that people will test their programs with a higher threshold with 3.13 and then when 3.14 release nears, we would have a better idea about how "aggressive" the default GC tuning should be.

Reply:
Returning an error `_PyGC_Init` results in this kind of error:
```
Fatal Python error: _PyGC_Init: Invalid PYTHON_GC_THRESHOLD value.
Python runtime state: preinitialized

Current thread 0x00007fe0505d6740 (most recent call first):
  <no Python frame>
```

Reply:
I think that makes sense. We do the same for other env vars.

Reply:
> I'm hoping this change can be included in the 3.13 RC and so I wanted it as simple and robust as possible.

Since we have already reached the `feature-freeze` stage and this is a new feature, it needs confirmation from @Yhg1s.

Reply:
That's right.  It's Thomas's call and I completely understand if he doesn't want it.  There has already been a bit too much excitement with the RCs.